gather.go

```diff
@@ -80,6 +81,7 @@ func (a *Agent) GatherCandidates() error {
 }
 
 func (a *Agent) gatherCandidates(ctx context.Context) {
```

Comment:
re: the XOR leak, it's not a leak so much as it's blocking closure since the underlying connection doing getXORMappedAddr must be forcefully closed. You could consider that out of scope of this change but it was adding a couple of seconds to failure cases IME.